src/views/ApartmentApp/components/StaffMenu/index.js

```diff
@@ -0,0 +1,174 @@
+import GetAppIcon from '@material-ui/icons/GetApp';
+import { DateTime } from 'luxon';
+import GordonLoader from '../../../../components/Loader';
+import housing from '../../../../services/housing';
+import ApplicationsTable from './components/ApplicationTable';
+
+/**
+ * @typedef { import('../../../../services/housing').ApplicationDetails } ApplicationDetails
+ * @typedef { import('../../../../services/housing').ApartmentApplicant } ApartmentApplicant
+ * @typedef { import('../../../../services/housing').ApartmentChoice } ApartmentChoice
+ */
+
+/**
+ * Renders the page for the apartment application housing staff menu
+ * @param {Object} props The React component props
+ * @param {*} props.authentication The user authentication
+ * @param {StudentProfileInfo} props.userProfile The student profile info of the current user
+ * @returns {JSX.Element} JSX Element for the staff menu web page
+ */
+const StaffMenu = ({ userProfile, authentication }) => {
+  const [loading, setLoading] = useState(true);
+
+  /** @type {[ApplicationDetails[], React.Dispatch<React.SetStateAction<ApplicationDetails[]>>]} ApplicationDetails */
+  const [applications, setApplications] = useState([]);
+
+  /** @type {[ApplicationDetails[], React.Dispatch<React.SetStateAction<ApplicationDetails[]>>]} Array of application details, after being formatted for use with react-csv */
+  const [applicationJsonArray, setApplicationJsonArray] = useState([]);
+
+  /** @type {[ApartmentApplicant[], React.Dispatch<React.SetStateAction<ApartmentApplicant[]>>]} Array of applicant info, after being formatted for use with react-csv */
+  const [applicantJsonArray, setApplicantJsonArray] = useState([]);
+
+  /** @type {[ApartmentChoice[], React.Dispatch<React.SetStateAction<ApartmentChoice[]>>]} Array of apartment choice info, after being formatted for use with react-csv */
+  const [apartmentChoiceJsonArray, setApartmentChoiceJsonArray] = useState([]);
+
+  const [dateStr, setDateStr] = useState('');
+  const filePrefix = 'apartapp';
+
+  /**
+   * Attempt to load all existing applications for the current semester
+   */
+  const loadAllCurrentApplications = useCallback(async () => {
+    setLoading(true);
+    let applicationDetailsArray = await housing.getAllApartmentApplications();
+    setApplications(applicationDetailsArray ?? []);
+    setLoading(false);
+  }, []);
+
+  useEffect(() => {
+    loadAllCurrentApplications();
+
+    // Generate string of today's date in ISO format for use in CSV filename
+    let date = new Date();
+    let dateStr = DateTime.fromJSDate(date).toISODate();
+    setDateStr(dateStr);
+  }, [userProfile, loadAllCurrentApplications]);
+
+  /**
+   * Generate arrays of objects to be converted to a CSV
+   * @param {ApplicationDetails[]} applicationDetailsArray an array of ApplicationDetails objects
+   */
+  const generateCSVData = useCallback((applicationDetailsArray) => {
```

Comment:
Similar to the above function, this is only called in the below `useEffect` and it also has no external dependencies. I don't see a need for it to use `useCallback`. You could put in inside the `useEffect` or just leave it externally, but either way `useCallback` isn't achieving much for you here.